src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java

```diff
@@ -130,17 +133,36 @@ public void accept(CtVisitor visitor) {
 		return findClass();
 	}
 
+	class F extends ClassLoader {
+
+	}
 	/**
-	 * Finds the class requested in {@link #getActualClass()}, using the
-	 * {@code ClassLoader} of the {@code Environment}
+	 * Finds the class requested in {@link #getActualClass()}.
+	 *
+	 * Looks for the class in the standard Java classpath, but also in the sourceClassPath given as option.
 	 */
 	@SuppressWarnings("unchecked")
 	protected Class<T> findClass() {
+		// first try the simple way
 		try {
-			return (Class<T>) getFactory().getEnvironment().getClassLoader().loadClass(getQualifiedName());
-		} catch (java.lang.ClassNotFoundException cnfe) {
+			return (Class<T>) Class.forName(getQualifiedName());
+		} catch (Throwable ignore) {
+		}
+
+		// then we also look into the sourceclasspth entries given
+		try {
+			String[] classpath = getFactory().getEnvironment().getSourceClasspath();
+			List<URL> classpathURL = new ArrayList<>();
+			for (String s : classpath) {
+				classpathURL.add(new File(s).toURL());
+			}
+			// creating a classloader on the fly is not the most efficient
+			// but it decreases the amount of state to maintain
+			// since getActualClass is only used in rare cases, that's OK.
+			return (Class<T>) new URLClassLoader(classpathURL.toArray(new URL[0])).loadClass(getQualifiedName());
+		} catch (Throwable e) {
 			throw new SpoonClassNotFoundException("cannot load class: " + getQualifiedName() + " with class loader "
-					+ Thread.currentThread().getContextClassLoader(), cnfe);
+					+ Thread.currentThread().getContextClassLoader(), e);
```

Comment:
Isn't it misleading to say "with class loader " Thread.currentThread().getContextClassLoader()? This is not this one that is in cause in here, but rather the URLClassLoader with class paths. Shouldn't you better say "in class paths ...".